src/linear_algebra/free_module.lean

```diff
@@ -334,6 +334,56 @@ lemma submodule.exists_is_basis_of_le_span
   ∃ (n : ℕ) (b : fin n → N), is_basis R b :=
 submodule.exists_is_basis_of_le le (is_basis_span hb)
 
+
+variable {M}
+
+/-- A finite type torsion free module over a PID is free. -/
+lemma module.free_of_finite_type_torsion_free [fintype ι] {s : ι → M} (hs : span R (range s) = ⊤)
```

Comment:
Note that `module.finite` exists already. Is there a reason that you use the explicit `s` instead in this statement?

Reply:
This version is the one that is nice to prove. My other PR #7343 is adding the missing glue to state the version you have in mind. When both those PRs will be merged I'll PR the version you have in mind.